db/internal_stats.cc

```diff
@@ -471,6 +471,28 @@ bool InternalStats::GetIntPropertyOutOfMutex(
   return (this->*(property_info.handle_int))(value, nullptr /* db */, version);
 }
 
+/*
+ * This function performs sanity check on the compaction stats. This function
+ * should be used in tests only.
+ *
+ * This function verifies that the number of compactions caused by different
+ * reasons sum up to equal the total number of compactions.
+ * @return true if sanity check passes, false otherwise
+*/
+bool InternalStats::TEST_CompactionStatsSanityCheck() const {
```

Comment:
is it possible to move this logic to the test file? I believe we usually use "TEST_*" functions to expose private members for convenience, not so much for the actual test logic.